README.md

```diff
@@ -202,3 +202,27 @@ If you don't like the way `fix_includes.py` munges your `#include` lines, you ca
 * If `fix_includes.py` has suggested a private header file (such as `<bits/stl_vector.h>`) instead of the proper public header file (`<vector>`), you can fix this by inserting a specially crafted comment near top of the private file (assuming you can write to it): '`// IWYU pragma: private, include "the/public/file.h"`'.
 
 Current IWYU pragmas are described in [IWYUPragmas](docs/IWYUPragmas.md).
+
+### Platform tweaks ###
+
+#### libc++ ####
+
+When analyzing C++ code on macOS, IWYU uses probing (implicitly via Clang) to find a libc++ standard library installation.
+
+If there is no installation of libc++ discoverable to Clang's probing, or an alternative libc++ is desired, users can specify relevant paths directly on command-line, e.g.:
+
+    # for Homebrew version of llvm-14
+    include-what-you-use \
+        -nostdinc++ \
+        -isystem /opt/homebrew/opt/llvm\@14/include/c++/v1 \
+        sourcefile.cc
+
+    # for Apple Command Line Tools version of libc++
+    include-what-you-use \
+        -nostdinc++ \
+        -isystem /Library/Developer/CommandLineTools/usr/include/c++/v1 \
+        sourcefile.cc
+
+The mechanics of this technique are described in more detail for Clang here: <https://libcxx.llvm.org//UsingLibcxx.html#using-a-custom-built-libc>.
```

Comment:
It may be worth mentioning how to find where xcode has installed the SDK:

`xcrun --sdk macosx --show-sdk-path`

appending `/usr/invlude/c++/v1` will get you to the C++ headers.

ref: https://stackoverflow.com/questions/18741675/how-to-get-the-path-of-latest-sdk-available-on-mac

LGTM though

Reply:
Nice, added.

Reply:
Hmm, I actually ran into a bit of a snag when I tried this. I have to use the following combination to IWYU anything with libc++ headers:
```
include-what-you-use \
      -isystem $(xcrun --sdk macosx --show-sdk-path)/usr/include \
      -isystem /Library/Developer/CommandLineTools/usr/include/c++/v1 \
      t.cc 
```

So the `xcrun` trick to get at the system libc, but a hardcoded CommandLineTools path to get the libc++ search path. Any other combo fails with weird conflicts.

Reply:
On my system `/Library/Developer/CommandLineTools/usr/include/c++/v1` is essentially empty. So I use the xcrun path for everything (?except the clang-resource-headers?).

FWIW I have my IWYU linked clang configured with `-DCMAKE_DEFAULT_SYSROOT=$(xcrun --sdk macos --show-sdk-path)`. However I'm using system clang to compile IWYU. CMake configures the IWYU build (that I'm using for testing) to include `-isysroot $(xcrun-path)` in compiler commands without me doing anything. So iwyu_tool puts that on my `include-what-you-use` lines, and everything seems to work.

That said, if I replace the `-isysroot` with `-isystem` for the libc and libc++ paths, things go wrong - I'm not sure why.

Reply:
Ah, good call on the `-isysroot`, I can get away with:
```
$ include-what-you-use -isysroot $(xcrun --show-sdk-path) --verbose tests/1293/t.cc
... cut for brevity ...
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/local/include"
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1
 /Users/kimgrasman/code/iwyu/out/main/lib/clang/18/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)
End of search list.
... cut for brevity ...
```

Reply:
Not sure how that plays with a brew-provided LLVM; it somehow needs to be combined with a compatible libc... Hm.

Reply:
After a bit of messing around, the following seems to work here:

```
include-what-you-use \
       -isystem $(xcrun --sdk macosx --show-sdk-path)/usr/include/c++/v1 \
       -isystem $(xcrun --sdk macosx --show-sdk-path)/usr/include \
       -isystem $(dirname $(xcrun --find clang))/../libclang/14.0.3/include \
       t.cc```